tests/std/tests/P2321R2_views_adjacent/test.cpp

```diff
@@ -0,0 +1,949 @@
+                assert(*i2 == expected[0]);
+            }
+        }
+
+        if constexpr (sized_sentinel_for<BI, BI>) { // Check differencing
+            same_as<range_difference_t<V>> auto diff = i - i;
+            assert(diff == 0);
+            assert((i - ranges::next(i)) == -1);
+            assert((ranges::next(i) - i) == 1);
+        }
+
+        if constexpr (sized_sentinel_for<sentinel_t<R>, I>) { // Check differencing with sentinel<not const>
+            const auto s                                    = r.end();
+            const auto size                                 = ranges::ssize(expected);
+            const same_as<range_difference_t<V>> auto diff1 = i - s;
+            assert(diff1 == -size);
+            const same_as<range_difference_t<V>> auto diff2 = s - i;
+            assert(diff2 == size);
+        }
+
+        if constexpr (sized_sentinel_for<sentinel_t<const R>, I>) { // Check differencing with const sentinel<const>
+            const auto s                                    = as_const(r).end();
+            const auto size                                 = ranges::ssize(expected);
+            const same_as<range_difference_t<V>> auto diff1 = i - s;
+            assert(diff1 == -size);
+            const same_as<range_difference_t<V>> auto diff2 = s - i;
+            assert(diff2 == size);
+        }
+
+        { // Check iter_move
+            same_as<repeated_tuple<iter_rvalue_reference_t<BI>, N>> decltype(auto) rval = iter_move(as_const(i));
+            assert(rval == expected[0]);
+            STATIC_ASSERT(noexcept(iter_move(i))
+                          == (noexcept(ranges::iter_move(declval<BI>()))
+                              && is_nothrow_move_constructible_v<iter_rvalue_reference_t<BI>>) );
+        }
+
+        if constexpr (indirectly_swappable<BI>) { // Check iter_swap
+            STATIC_ASSERT(is_void_v<decltype(ranges::iter_swap(as_const(i), as_const(i)))>);
+            STATIC_ASSERT(
+                noexcept(iter_swap(i, i)) == noexcept(ranges::iter_swap(declval<const BI&>(), declval<const BI&>())));
+            // Note: other tests are defined in 'test_iter_swap' function
+        }
+    }
+
+    // Check adjacent_view::iterator<const>
+    if constexpr (CanMemberBegin<const R>) {
+        using CBI = iterator_t<const V>;
+        using CI  = iterator_t<const R>;
+        STATIC_ASSERT(forward_iterator<CI>);
+
+        // Check iterator_category
+        STATIC_ASSERT(same_as<typename CI::iterator_category, input_iterator_tag>);
+
+        // Check iterator_concept
+        using IterConcept = typename CI::iterator_concept;
+        STATIC_ASSERT(random_access_range<V> == same_as<IterConcept, random_access_iterator_tag>);
+        STATIC_ASSERT(
+            (bidirectional_range<V> && !random_access_range<V>) == same_as<IterConcept, bidirectional_iterator_tag>);
+        STATIC_ASSERT((forward_range<V> && !bidirectional_range<V>) == same_as<IterConcept, forward_iterator_tag>);

```

Comment:
This is saying `<V>` but I suspect it should be saying `<const V>` even if the answer happens to be the same.